.github/workflows/cpp.yml

```diff
@@ -146,8 +146,7 @@ jobs:
       ARROW_WITH_SNAPPY: ON
       ARROW_WITH_ZLIB: ON
       ARROW_WITH_ZSTD: ON
-      # System Abseil installed by Homebrew uses C++ 17
-      CMAKE_CXX_STANDARD: 17
+      GTest_SOURCE: BUNDLED
```

Comment:
Why do we need this? Do we need newer GTest for `std::string_view`? If so, how about updating required GTest version too? https://github.com/apache/arrow/blob/master/cpp/cmake_modules/ThirdpartyToolchain.cmake#L2078

Reply:
The GTest version does not need to be new, but GTest needs to be compiled in C++17 mode otherwise some test helpers are needed (such as equality between string_views).

Reply:
Ah, I see.